src/types/cursor/xcursor_manager.rs

```diff
@@ -0,0 +1,106 @@
+use wlroots_sys::{wlr_xcursor_manager, wlr_xcursor_manager_create, wlr_xcursor_manager_load,
+                  wlr_xcursor_manager_set_cursor_image, wlr_xcursor_manager_destroy,
+                  wlr_xcursor_manager_get_xcursor, wlr_xcursor_manager_theme};
+use types::{Cursor, XCursor, XCursorTheme};
+use std::ptr;
+use utils::safe_as_cstring;
+use std::marker::PhantomData;
+
+#[derive(Debug)]
+pub struct XCursorManagerTheme<'manager> {
+    theme: *mut wlr_xcursor_manager_theme,
+    phantom: PhantomData<XCursorManager<'manager>>
+}
+
+#[derive(Debug)]
+pub struct XCursorManager<'manager> {
+    manager: *mut wlr_xcursor_manager,
+    phantom: PhantomData<XCursor<'manager>>
+}
+
+impl<'manager> XCursorManagerTheme<'manager> {
+    pub(crate) fn new(theme: *mut wlr_xcursor_manager_theme) -> Self {
+        XCursorManagerTheme {
+            theme: theme,
+            phantom: PhantomData
+        }
+    }
+
+    pub fn scale(&self) -> f32 {
+        unsafe {
+            (*self.theme).scale
+        }
+    }
+
+    pub fn theme(self) -> XCursorTheme {
+        unsafe {
+            XCursorTheme::new((*self.theme).theme)
+        }
+    }
+}
+
+impl<'manager> XCursorManager<'manager> {
+    pub fn create<T: Into<Option<String>>>(name: T, size: u32) -> Option<Self> {
+        unsafe {
+            let name_str = name.into().map(safe_as_cstring);
+            let name_ptr = name_str.map(|s| s.as_ptr()).unwrap_or(ptr::null_mut());
+            let manager = wlr_xcursor_manager_create(name_ptr, size);
+            if manager.is_null() {
+                None
+            } else {
+                Some(XCursorManager { manager: manager, phantom: PhantomData })
+            }
+        }
+    }
+
+    pub fn size(&self) -> u32 {
+        unsafe { (*self.manager).size }
+    }
+
+    pub fn get_xcursor<T: Into<Option<String>>>(&'manager self, name: T, scale: f32) -> Option<XCursor<'manager>> {
```

Comment:
Now this might be mis-understanding the lifetime of the xcursor correctly, so if I'm wrong please correct me.

Could you not make the lifetime only apply to `get_xcursor`? E.g there's only a lifetime constraint when you call this function because only this actually gets the xcursor?

This would make the signature look like this instead (and the `'manager` lifetime annotation would no longer be on the struct and you no longer need the `PhantomData`)

```rust
pub fn get_xcursor<'manager, T: Into<Option<String>>>(&'manager self, name: T, scale: f32) -> Option<XCursorManager<'manager>> {
    ... 
}